crm_claim_ref_smartbutton/models/base_model.py

```diff
@@ -0,0 +1,111 @@
+            if sheets:
+                sheets[0].insert(0, button_box)
+            else:
+                forms = eview.xpath("//form")
+                if forms:
+                    forms[0].insert(0, button_box)
+        else:
+            button_box = button_boxes[0]
+        field_name = 'crm_claim_linked_count_tech'
+        new_field = etree.Element(
+            'field', {'string': _('Linked claims'),
+                      'name': field_name,
+                      'widget': 'statinfo'})
+        new_button = etree.Element(
+            'button', {'class': 'oe_inline oe_stat_button',
+                       'type': 'object',
+                       'name': 'action_open_linked_claims_tech',
+                       'icon': 'fa-comments',
+                       'groups': 'base.group_sale_salesman'})
+        new_button.append(new_field)
+        button_box.append(new_button)
+        res['arch'] = etree.tostring(eview)
+        res['fields'][field_name] = self.fields_get([field_name])
+    return res
+
+
+@api.multi
+def _open_related_claims(self):
+    claim_obj = self.env['crm.claim']
+    act_window_obj = self.env['ir.actions.act_window']
+    search_term = '%s,%i' % (self._model, self.id)
+    claims = claim_obj.search([('ref', '=', search_term)])
+    result = act_window_obj.for_xml_id('crm_claim', 'crm_case_categ_claim0')
+    result['context'] = {'default_ref': search_term}
+    result['domain'] = [('id', 'in', claims.ids)]
+    return result
+
+
+def _add_magic_fields_wrapper(wrapped_func):
+    wrapped_func = wrapped_func.__func__
+
+    def _w(*args, **kwargs):
+        """Inject field in each model."""
+        cls = args[0]
+        name = 'crm_claim_linked_count_tech'
+        if name not in cls._fields:
+            cls._add_field(
+                name, fields.Integer(compute='count_linked_claim_tech',
+                                     automatic=True))
+        wrapped_func(*args, **kwargs)
+
+    return classmethod(_w)
+
+
+@api.one
+def count_linked_claim_tech(self):
+    self.crm_claim_linked_count_tech = self.env['crm.claim'].search_count(
+        [('ref', '=', '%s,%i' % (self._model, self.id))])
+
+
+# This should be monkey-patched always, or the computed field won't be
+# added to all models due to the module loading order
+models.BaseModel._add_magic_fields = _add_magic_fields_wrapper(
+    models.BaseModel._add_magic_fields)
+models.BaseModel.count_linked_claim_tech = count_linked_claim_tech
+
+
+class CrmClaimMonkeyPatcher(models.Model):
+    _name = 'crm.claim.monkey.patcher'
+    _auto = False  # doesn't create table in database
+
+    def _register_hook(self, cr):
+        # Monkey patch functions - Only when the module is installed
+        models.BaseModel.action_open_linked_claims_tech = _open_related_claims
+        models.BaseModel.base_fields_view_get = base_fields_view_get
+        models.BaseModel.fields_view_get = fields_view_get_crm_claim
+        registered_dbs.append(cr.dbname)
```

Comment:
Although monkey-patching is dirty by definition, this method patches all databases in the instance. It seems you use `registered_dbs` to keep track of that, but there's no need.

You can use [`_patch_method` and `_revert_method`](https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L5210-L5247); those will take care of this with you. It's a bit complex but you can [inspire yourself by how base_field_validator does it](https://github.com/OCA/server-tools/blob/8.0/base_field_validator/models/ir_model.py#L34-L83).